logic/transaction.js

```diff
@@ -525,6 +529,12 @@ class Transaction {
 			}
 		}
 
+		// Sanitize signatures property if none were found at this point
+		transaction.signatures =
```

Comment:
Array.
```suggestion
		transaction.signatures = Array.isArray(transaction.signatures) ? transaction.signatures || [];
```

Reply:
@MaciejBaj prettier made the change

Reply:
this.ready is doing the same check. Can we rely on it?

```
Multisignature.prototype.ready = function(transaction, sender) {
	if (!Array.isArray(transaction.signatures)) {
		return false;
	}

```

Reply:
@MaciejBaj will check though my understanding was that for all transactions it should be present as per Diego's comments; @diego-G can you confirm this?

Reply:
I would prefer to move `transaction.signatures` assignment to both transfer.js and multignature.js `objectNormalize` methods. If you sanitize the signatures outside of the normalize functions it introduces inconsistency.

Reply:
Yes, I confirm. Just bear in mind you mutate signatures too late. It has to be normalise/sanitise before ready. I would suggest to perhaps moving it to objectNormalise in transactions. What do you think @4miners ?

Reply:
@MaciejBaj @diego-G moved to objectNormalized in transactions.

Reply:
thanks!